src/lab16.py

```diff
@@ -288,15 +315,29 @@ def __init__(self, node, parent, previous, frame):
         self.previous = previous
         self.frame = frame
 
-        self.children = ProtectedField(node, "children", self.parent)
-        self.zoom = ProtectedField(node, "zoom", self.parent)
-        self.width = ProtectedField(node, "width", self.parent)
-        self.height = ProtectedField(node, "height", self.parent)
-        self.x = ProtectedField(node, "x", self.parent)
-        self.y = ProtectedField(node, "y", self.parent)
+        self.children = ProtectedField(self, "children", self.parent)
+        self.zoom = ProtectedField(self, "zoom", self.parent)
+        self.width = ProtectedField(self, "width", self.parent)
+        self.height = ProtectedField(self, "height", self.parent)
+        self.x = ProtectedField(self, "x", self.parent)
+        self.y = ProtectedField(self, "y", self.parent)
 
         self.has_dirty_descendants = True
 
+        # self.children has dependencies on the zoom and font style of various
+        # inline children. But we don't know which until self.recurse is
+        # called, and building up the dependencies would amount to a duplicate
+        # implementation of self.recurse.
```

Comment:
Then do I understand correctly that there are now two modes? Most fields have "frozen" dependencies, but a few don't?

Reply:
The `style` fields only depend on their parents, right? (For resolving computed styles and for inheritance.)